### PR TITLE
docs: convert BACKERS.md to backers.txt

### PR DESCRIPTION
**Pull request by dundargoc**
There is no reason for this file to be in project root, which is crowded
as is. This also fits nicely part of the ongoing work towards gathering
as much of the documentation as possible into one place.

### CODE REVIEW

File: runtime/doc/backers.txt [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
-Thank you to everyone who backed our [Bountysource fundraiser](https://www.bountysource.com/teams/neovim/fundraiser)!
 
-### Your name and URL in BACKERS.md.
+                     NVIM REFERENCE MANUAL
+
+
+==============================================================================
+Fundraiser Backers
+
+Thank you to everyone who backed the original Neovim Fundraiser
```
Comment (zeertzjq) [inline runtime/doc/backers.txt:10]: No punctuation at the end?

Comment (dundargoc) [reply runtime/doc/backers.txt:10]: Punctuation is an outdated concept.

**Comment by clason** [comment]
Turns out one of the tests (`functional/editor/completion_spec.lua:896`) uses `BACKERS.md`... 🫠

**Review by clason — APPROVED** [review]

File: runtime/doc/backers.txt [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
-Thank you to everyone who backed our [Bountysource fundraiser](https://www.bountysource.com/teams/neovim/fundraiser)!
 
-### Your name and URL in BACKERS.md.
+                     NVIM REFERENCE MANUAL
+
+
+==============================================================================
+Fundraiser Backers
+
+Thank you to everyone who backed the original Nvim Fundraiser.
```
Comment (clason) [inline runtime/doc/backers.txt:10]: I think this should be "Neovim" still? (Project, not editor?)

Comment (dundargoc) [reply runtime/doc/backers.txt:10]: Lol the test failed, and I was feeling lazy. I'll take a closer look to see if there's a way to ignore a file.

Comment (dundargoc) [reply runtime/doc/backers.txt]: Adjusted spell script a bit to allow ignoring files.

Comment (clason) [reply runtime/doc/backers.txt]: Ah, I see. Not worth messing with the scripts over this, then.

**Pull request merged**